hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassAlertsServiceImpl.java

```diff
@@ -1482,14 +1480,39 @@ private void handleResolveOptions(String tenantId, String triggerId, boolean che
 
     @Override
     public void sendData(Data data) throws Exception {
-        alertsEngine.sendData(data);
+        sendData(Collections.singleton(data));
     }
 
     @Override
     public void sendData(Collection<Data> data) throws Exception {
+        checkDataDrivenGroupTriggers(data);
         alertsEngine.sendData(data);
     }
 
+    private void checkDataDrivenGroupTriggers(Collection<Data> data) throws Exception {
+        if (!dataDrivenGroupCacheManager.isCacheActive()) {
+            return;
+        }
+        for (Data d : data) {
+            if (isEmpty(d.getSource())) {
```

Comment:
I wonder if we should check at this level is Data.source == __none__ as well.
If I understood the design, instead to use null/empty value (perhaps in other part of the code I can review it) we use that string constant. So, as checkDataDrivenGroupTriggers() will be executed very often, perhaps this optimitization would be fine.